### ISSUE TITLE
SM18: Error: "Cannot open player device"

### ISSUE THREAD

**Issue by Golddouble** [alessivs/supermemo-wine#33]
I have tried to install SuperMemo 18 Lutris variant.
Everything went as expected, but the sound does not work:
Message: "Cannot open player devic"

How did I install SM 18?

System: Linux MX 19.4
1. started Lutris 0.5.7.1
2. searched for supermemo "search Lutis net"
3. selected SM 18
4. pressed the install button
5. provided the folder for the prefix and the path to the sm18 installer

Started sm18 with Lutris "play"

What I have tried to solve the issue:
Pressed the configure button in Lutris.
In the tab "runner options" I have added this:
![ksnip_20220111-160445](https://user-images.githubusercontent.com/13977359/148967493-b77197f7-1e23-4535-95e8-7c7baeb5f791.jpg)

Then I have tried again.
-> Same message

After I have started Winetricks from Lutris and installed "quartz".

Tried again. Still the same.

After I have change the wine version in Lutris "Runner options".
I have tried:
-Lutris-5.0
-Wine HQ (Stagging (6.5)
-System (6.5 (Stagging))

It's still the same.

I had the same issue, when I was installing SM 17 1.5 years ago.
With your help I was then successful. You have said, that you will make a Lutris install variant. I took it for granted, that in this Lutris Install the fix for "sound" is already included, because it takes another wine (Lutris wine) that contains all what is necessary to also bring the sound to work. (?)

Isn't this so?

Thank you.

**Comment by alessivs** [comment]
What kind of sound is it? Is it a sound component with only the playback functionality?

If you can, also locate the sound file. Let's say it's called `42.mp3`. I'd like the output of the following command, to distinguish between codec issues and playback issues:

`mediainfo 42.mp3`

Notes:

- Sound troubleshooting is more reliable when performed on a native installation (thus removing the virtualization variable).
- Running the same prefix under a different version of Wine may trigger an in-place upgrade of the prefix; some system file could be replaced or recreated and may introduce an issue.
- Installers (or their installation outcomes) remain largely untested on recent Wine 6.x releases.

**Comment by Golddouble** [comment]
> What kind of sound is it? Is it a sound component with only the playback functionality?
> 
It are simple *.mp3 files. For example 00122q.mp3
They are connected to a sound component. And I only want to play and hear them.

> If you can, also locate the sound file. Let's say it's called `42.mp3`. I'd like the output of the following command, to distinguish between codec issues and playback issues:
>
> `mediainfo 42.mp3`

I do not kow, where I have to run this command.
I have tried in Linux. (Command does not exist) and I have tried in windows Dos prompt (not wine) -> Command does not exist.

> Notes:
> 
>Sound troubleshooting is more reliable when performed on a native installation (thus removing the virtualization variable).

Do I understand correctly that I should undo the change I documented with the picture above?

>Running the same prefix under a different version of Wine may trigger an in-place upgrade of the prefix; some system file could be replaced or recreated and may introduce an issue.

It has not worked with the Lutris-4.21 wine version, only then I have tried out the other versions.
.
.
.
.
**Edit:** 
Please let me add the following info:
I have used the same collection in SM 17. And sound is working there.

But that was also a very difficult birth at the time, but finally successful.
I have used a *.yaml file at that time you have made for me.
See here: https://github.com/alessivs/supermemo-wine/issues/10#issuecomment-633610372
```
wget https://is.gd/ii5l6W -O supermemo-17-winaudio.yaml
lutris -i supermemo-17-winaudio.yaml
```

Now I have installed it directly via Lutris, but SM18 instead of SM17.
I was thinking, that all innovation for sound is now also in this Lutris-SM18. Isn't it? Or was that too specific?

Additional Info:
And I have changed this in the new Lutris-SM18:
![ksnip_20220111-160445](https://user-images.githubusercontent.com/13977359/149027553-b6189775-671c-4878-b7a0-601b0305e4dc.jpg)
Because I have this in the Lutris-SM17. So I though, that this is what is missing. But I was wrong, as already mentioned above.

Thank you.

**Comment by Golddouble** [comment]
I guess I have found the supermemo-17-winaudio.yaml:
[supermemo-17-winaudio (letzte für guten Audioklang).yaml.zip](https://github.com/alessivs/supermemo-wine/files/7850139/supermemo-17-winaudio.letzte.fur.guten.Audioklang.yaml.zip)

Maybe you can make something like this also for Lutris-SM18?

**Comment by alessivs** [comment]
## Installer

Sorry for the lack of time. This is untested for now: [supermemo-18-winaudio.yaml.gz](https://github.com/alessivs/supermemo-wine/files/7850564/supermemo-18-winaudio.yaml.gz)

## Addendum (on mediainfo)

> I do not kow, where I have to run this command.

Don't worry. I interpret it's your same situation as before, where you need the native LAVF codecs. Understood; mediainfo might not be needed.

In any case, for `mediainfo` you just do:

```sh
sudo apt install mediainfo
```

(there is also a gui in the package `mediainfo-gui`).

And then you can simply invoke `mediainfo /path/to/media/file` to query the information of a media file.

**Comment by Golddouble** [comment]
Thank you for the SM18 *.yaml !

Yes, it seems to work so far.
Very pleased to have this good news.

**Comment by Golddouble** [comment]
There used to be these `compact_navigation_bar.smXXXX.delta` files, to fix the broken bar list. Unfortunately, I no longer found them here:
https://github.com/alessivs/supermemo-miscellany/blob/master/readme.org

I have looked for them in `https://github.com/alessivs/supermemo-wine` too. But I have no longer found them.

(?)

**Comment by alessivs** [comment]
Yes, that repository was only a holder of files and not part of the installer initiative. The last patch fixing the navbar was produced for SM 18.04, then I stopped using v18 personally. The patch update for 18.05 is planned. Will reply to this issue when there's news.

**Comment by Golddouble** [comment]
OK. Thank you for your kind help.
